### PR TITLE
JSON followup fixes

### LINKED ISSUE #6486: JSON followup issues
Issue overview
--------------
- [ ] Visual Studio dependency issues - GenerateEmbeddedEpJSONSchema always appears out of date, thinks for about 10 seconds then does nothing with it and moves on with the rest of the build.
- [ ] Better error messages for input errors. (see more details below).

An ill formed input file terminates via the idfParser without any indication of why. Users won't know why their input file is failing. Some form of message should report the object name and possibly the input field name expected, or something along those lines to give users direction of where to look.

Take 1ZoneEvapCooler and change this:

    ZoneHVAC:EquipmentList,
      Main Zone Equipment,     !- Name
      SequentialLoad,          !- Load Distribution Scheme
      AirTerminal:SingleDuct:Uncontrolled,  !- Zone Equipment 1 Object Type
      Main Zone Direct Air,    !- Zone Equipment 1 Name
      1,                       !- Zone Equipment 1 Cooling Sequence
      2,                       !- Zone Equipment 1 Heating or No-Load Sequence
      ZoneHVAC:Baseboard:Convective:Electric,  !- Zone Equipment 2 Object Type
      Main Zone Baseboard,     !- Zone Equipment 2 Name
      2,                       !- Zone Equipment 2 Cooling Sequence
      1;                       !- Zone Equipment 2 Heating or No-Load Sequence

to this:

    ZoneHVAC:EquipmentList,
      Main Zone Equipment,     !- Name
      AirTerminal:SingleDuct:Uncontrolled,  !- Zone Equipment 1 Object Type
      Main Zone Direct Air,    !- Zone Equipment 1 Name
      1,                       !- Zone Equipment 1 Cooling Sequence
      2,                       !- Zone Equipment 1 Heating or No-Load Sequence
      ZoneHVAC:Baseboard:Convective:Electric,  !- Zone Equipment 2 Object Type
      Main Zone Baseboard,     !- Zone Equipment 2 Name
      2,                       !- Zone Equipment 2 Cooling Sequence
      1;                       !- Zone Equipment 2 Heating or No-Load Sequence

If a user copy\pastes an object from an old file (or makes any other simple mistake) and runs it, they won't have a clue as to why it failed.

Error file results:

  Program Version,EnergyPlus, Version 8.9.0-1c5ba897d1, YMD=2018.02.21 11:48,
    ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
    ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
    ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
    ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
    ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 0 Severe Errors; Elapsed Time=00hr 00min  0.21sec

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mbadams5**
Pull request overview
---------------------
This PR fixes #6486 validation issues and a few other issues lingering from the initial pull request.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: datasets/RooftopPackagedHeatPump.idf [hunk 1]
```diff
@@ -155,7 +155,7 @@
     1.25,                    !- Maximum Value of x
     0.9275,                  !- Minimum Curve Output
     1.0198,                  !- Maximum Curve Output
-    Flow Fraction,           !- Input Unit Type for X
+    VolumetricFlow,          !- Input Unit Type for X
```
Comment (mjwitte) [inline datasets/RooftopPackagedHeatPump.idf]: These should all be `Dimensionless`, not VolumetricFlow.

**Comment by Myoldmopar** [comment]
@mbadams5 Please go ahead and address these, pull in latest develop, and push those changes up for final review.  This is the last PR for 8.9.0 barring a showstopper found in release testing.

**Comment by mjwitte** [comment]
@mbadams5 @Myoldmopar Running a v8.8 5ZoneAirCooled with this branch, it fails as expected, because the new field in ZoneHVAC:Equipment list is missing.
```
   ** Severe  ** <root>[ZoneHVAC:EquipmentList][SPACE1-1 Eq][equipment][0][zone_equipment_object_type] - Failed to match against any enum values.
   **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.
```
The error message is a bit cryptic, although most users will likely be able to figure it out. 

Also, in this case, the warning about version mismatch does not get written. The version warning is still in there, from v8.8 1ZoneUncontrolled
```
   ** Warning ** Version: in IDF="8.8" not the same as expected="8.9"
```
Any chance this could be checked and written when there are input errors?

Tried another file with multiple problems and got this err output:
```
   ** Warning ** Line: 220 Index: 17 - "MaterialSayWhat" is not a valid Object Type. Attempting to skip object.
   ** Warning ** Line: 220 Index: 17 - "MaterialSayWhat" is not a valid Object Type. Attempting to skip object.
   ** Severe  ** <root>[Site:Location][Denver Centennial  Golden   N_CO_USA Design_Conditions][latitude] - Expected number less than or equal to 90.000000
   ** Severe  ** <root>[SizingPeriod:DesignDay][Denver Centennial  Golden   N Ann Htg 99% Condns DB][day_type] - Failed to match against any enum values.
```
If you make changes, I'd say Severe instead of Warning for invalid object type.
Also, can you add the bad field value to the error message?

**Comment by mjwitte** [comment]
Here's the file from that last one 
[1ZoneUncontrolled-RangeCheckAndMore-8.8.idf.txt](https://github.com/NREL/EnergyPlus/files/1833226/1ZoneUncontrolled-RangeCheckAndMore-8.8.idf.txt)

File: src/EnergyPlus/InputProcessing/IdfParser.cc [hunk 2]
```diff
@@ -201,8 +201,18 @@ json IdfParser::parse_idf( std::string const & idf, size_t & index, bool & succe
 			auto const parsed_obj_name = parse_string( idf, index, success );
 			auto const obj_name = normalizeObjectType( parsed_obj_name );
 			if ( obj_name.empty() ) {
-				warnings_.emplace_back( "Line: " + std::to_string( cur_line_num ) + " Index: " + std::to_string( index_into_cur_line ) +
-										" - \"" + parsed_obj_name + "\" is not a valid Object Type. Attempting to skip object." );
+				errors_.emplace_back( "Line: " + std::to_string( cur_line_num ) + " Index: " + std::to_string( index_into_cur_line ) +
```
Comment (mjwitte) [inline src/EnergyPlus/InputProcessing/IdfParser.cc]: @mbadams5 Thanks for the changes. For this case, it now stops after this without reporting any other input error and without the version mismatch warning. So, maybe this was better as it was before or you still report Severe but continue processing.

**Comment by Myoldmopar** [comment]
I've reproduced these locally and am starting to address them.  So far they appear to be valid problems with the unit tests (duplicate names in idf snippets and datasets).  I'll try to keep track of the ones I've fixed here so no work is duplicated.  

- [X] Piping System Full Simulation
- [X] Datasets chillers

**Comment by mjwitte** [comment]
@mbadams5 When input comes in with idf does the json equivalent input file get produced? If so, what's it called and can we make a quick change to run scripts to save it with the rest of the  output?

**Comment by mbadams5** [comment]
@mjwitte Yes it can output the generated epJSON but does not by default. [From the command line you can use the `-c` flag to output the generated epJSON or generate IDF from epJSON input.](https://github.com/NREL/EnergyPlus/blob/267b8ef59d1b4b12333261415ae0fb42fcee85b9/src/EnergyPlus/CommandLineInterface.cc#L152) I don't know which run scripts you want to change.

**Comment by mjwitte** [comment]
I'm primarily thinking of EP-Launch on windows. I don't know what the run scripts on other platforms do. Given that it's not produced without the command line option, and given the timing, we should probably just leave as-is for now and support it as an option in EP-Launch3 for all platforms. Advanced users who want it can modify their own scripts to generate the file.

**Comment by Myoldmopar** [comment]
FINAL FAILING UNIT TEST :rofl:

**Comment by Myoldmopar** [comment]
OK, Windows is happy, and Linux is happy except for the dumb bad choice fields in those performance test files.  I'm going to push those fixes up and merge this in.  I'm then going to tag RC1.

**Pull request merged**